tests/test_mariadb.py

```diff
@@ -208,3 +195,43 @@ def wait_for_server():
     assert rows and len(rows) == 2
     _, num, data = rows[-1].split()
     assert num == "100" and data == "abcdef"
+
+
+def test_mariadb_healthcheck_innodb_initialized(auto_container_per_test):
+    """
+    Test if InnoDB (storage engine) has completed initializing.
+
+    See: https://mariadb.com/kb/en/using-healthcheck-sh/#-innodb_initialized
+    """
+    conn = auto_container_per_test.connection
+
+    _wait_for_server(conn)
+
+    conn.check_output("healthcheck.sh --su-mysql --innodb_initialized")
+
+
+def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
+    """
+    Test if Galera Cluster (multi-primary cluster) is disabled (experimental feature).
+
+    See: https://mariadb.com/kb/en/using-healthcheck-sh/#-galera_online
+    """
+    conn = auto_container_per_test.connection
+
+    _wait_for_server(conn)
+
+    conn.run_expect([1], "healthcheck.sh --su-mysql --galera_online")
+
+
+def test_mariadb_query_innodb_initialized(auto_container_per_test):
+    """
+    Test if InnoDB (storage engine) has completed initializing.
+    """
+    conn = auto_container_per_test.connection
+
+    _wait_for_server(conn)
+
+    # the query is similar to: healthcheck.sh --innodb_initialized
+    conn.check_output(
+        f"mariadb -p{MARIADB_ROOT_PASSWORD} -e \"select 1 from information_schema.ENGINES WHERE engine='innodb' AND support in ('YES', 'DEFAULT', 'ENABLED')\""
+    )
```

Comment:
Why are you adding this test case if it performs the same action as `healthcheck.sh --innodb_initialized`?

Reply:
The main idea was to check if the healthcheck.sh script ever changes behavior.

Reply:
I am not terribly enthusiastic about this kind of test to be honest. We'll be testing implementation details which I'd consider a bit of an anti-pattern. I'd add this if we had a regression involving this, but I'm not aware of any. However, I'm open to other points of view.

Reply:
I don't mind removing it. In theory, if the behavior of healthcheck.sh changes, the test are very likely to fail.